.github/workflows/docker-connector-base-image-tests.yml

```diff
@@ -0,0 +1,285 @@
+          # build-args: |
+          #   BASE_IMAGE=...
+
+      - name: Debug outputs
+        run: |
+          echo "python-base-image=${{ steps.docker-build-python-base.outputs.imageid }}"
+
+      - name: Run Python Base Image Vulnerability Scan
+        uses: anchore/scan-action@v6
+        with:
+          image: "ghcr.io/airbytehq/python-connector-base:draft-pr-${{ github.event.pull_request.number }}"
+          output-format: "table"
+          severity-cutoff: critical # TODO: revert to 'high'
+          fail-build: true
+
+    outputs:
+      python-base-image: ${{ steps.docker-build-python-base.outputs.imageid }}
+
+  test-java-connector-image-matrix:
+    runs-on: ubuntu-latest
+    needs: [detect-changes, build-java-base-image]
+    if: >
+      needs.detect-changes.outputs.java-connector
+    strategy:
+      fail-fast: false
+      matrix:
+        connector:
+          - source-postgres
+          - destination-postgres
+          - destination-snowflake
+    name: Build ${{ matrix.connector }} Java Test Image
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      # Java deps
+      - uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: 21
+          cache: gradle
+
+      - uses: gradle/actions/setup-gradle@v4
+        with:
+          cache-read-only: false
+          cache-write-only: false
+          add-job-summary: "on-failure"
+
+      - name: Build Connector Tarball
+        run: |
+          ./gradlew :airbyte-integrations:connectors:${{ matrix.connector }}:distTar
+
+      - name: Log in to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Debug BUILD_ARGS
+        run: |
+          echo "BASE_IMAGE=${{ needs.build-java-base-image.outputs.java-base-image }}"
+          echo "CONNECTOR_NAME=${{ matrix.connector }}"
+
+      - name: Build Java Connector Image
+        id: docker-build-java-connector
+        uses: docker/build-push-action@v5
+        with:
+          context: airbyte-integrations/connectors/${{ matrix.connector }}
+          file: docker-images/Dockerfile.java-connector
+          platforms: linux/amd64,linux/arm64
+          tags: ghcr.io/airbytehq/${{ matrix.connector }}:draft-pr-${{ github.event.pull_request.number }}
+          build-args: |
+            BASE_IMAGE=ghcr.io/airbytehq/java-connector-base:draft-pr-${{ github.event.pull_request.number }}
+            CONNECTOR_NAME=${{ matrix.connector }}
+          push: true
+
+      - name: Run `spec` Image Test
+        run: |
+          docker run \
+            --rm \
+            ghcr.io/airbytehq/${{ matrix.connector }}:draft-pr-${{ github.event.pull_request.number }} \
+            spec
```

Comment:
and I believe this should be eventually replaced with a test step?

Reply:
@davinchia - As of now, this runs `spec` on the image. We could go further and test `check` but we'll need to grab credentials first, which this PR doesn't yet try to tackle.

Reply:
Yeap. I resolved my comment since I realized this was really for the other PR I was reviewing, although I think my point still stands. Eventually we need to go through the entire test suite instead of only one command. You're right; this is a new workflow, and we don't have to block on this.